tests/unit/test_config.py

```diff
@@ -1,209 +1,80 @@
 
+        assert config.API_KEY == api_key
+        assert config.HOST == controller_host
+        assert config.OPENAPI_CONFIG == openapi_config
 
-def test_init_with_positional_args():
-    api_key = "my-api-key"
-    environment = "test-env"
-    host = "my-controller-host"
-    log_level = None  # deprecated property but still in positional list
-    openapi_config = OpenApiConfiguration(api_key="openapi-api-key")
+    def test_init_with_mispelled_kwargs(self, caplog):
+        PineconeConfig(api_key='my-api-key', unknown_kwarg='bogus')
+        assert "__init__ had unexpected keyword argument(s): unknown_kwarg" in caplog.text
 
-    pinecone.init(api_key, host, environment, log_level, openapi_config)
-
-    assert Config.API_KEY == api_key
-    assert Config.ENVIRONMENT == environment
-    assert Config.CONTROLLER_HOST == host
-    assert Config.OPENAPI_CONFIG == openapi_config
-
-
-def test_init_with_kwargs():
-    env = "test-env"
-    api_key = "my-api-key"
-    controller_host = "my-controller-host"
-    openapi_config = OpenApiConfiguration(api_key="openapi-api-key")
-
-    pinecone.init(api_key=api_key, environment=env, host=controller_host, openapi_config=openapi_config)
-
-    assert Config.API_KEY == api_key
-    assert Config.ENVIRONMENT == env
-    assert Config.CONTROLLER_HOST == controller_host
-    assert Config.OPENAPI_CONFIG == openapi_config
-
-
-def test_init_with_mispelled_kwargs(caplog):
-    pinecone.init(invalid_kwarg="value")
-    assert "init had unexpected keyword argument(s): invalid_kwarg" in caplog.text
-
-
-def test_init_with_file_based_configuration():
-    """Test that config can be loaded from a file"""
-    env = "ini-test-env"
-    api_key = "ini-api-key"
-    controller_host = "ini-controller-host"
-
-    with tempfile.NamedTemporaryFile(mode="w") as f:
-        f.write(
-            f"""
-        [default]
-        environment: {env}
-        api_key: {api_key}
-        controller_host: {controller_host}
+    def test_resolution_order_kwargs_over_env_vars(self):
         """
-        )
-        f.flush()
-
-        pinecone.init(config=f.name)
-
-    assert Config.API_KEY == api_key
-    assert Config.ENVIRONMENT == env
-    assert Config.CONTROLLER_HOST == controller_host
-
-
-def test_resolution_order_kwargs_over_env_vars():
-    """
-    Test that when config is present from multiple sources,
-    the order of precedence is kwargs > env vars
-    """
-    os.environ["PINECONE_ENVIRONMENT"] = "env-var-env"
-    os.environ["PINECONE_API_KEY"] = "env-var-api-key"
-    os.environ["PINECONE_CONTROLLER_HOST"] = "env-var-controller-host"
-
-    env = "kwargs-env"
-    api_key = "kwargs-api-key"
-    controller_host = "kwargs-controller-host"
-
-    pinecone.init(environment=env, api_key=api_key, host=controller_host)
-
-    assert Config.API_KEY == api_key
-    assert Config.ENVIRONMENT == env
-    assert Config.CONTROLLER_HOST == controller_host
-
-
-def test_resolution_order_kwargs_over_config_file():
```

Comment:
It looks like we lost the tests around resolution order over a config file, but I'd assume that's fine given we'll need to add new testing around the config singleton and stuff anyways, and the config file setup isn't really documented.